bsp/esp_bsp_generic/include/bsp/esp_bsp_generic.h

```diff
@@ -0,0 +1,277 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ESP BSP: Generic
+ */
+
+#pragma once
+
+#include "sdkconfig.h"
+#include "driver/gpio.h"
+#include "driver/i2c.h"
+#include "driver/sdmmc_host.h"
+#include "iot_button.h"
+#include "led_indicator.h"
+
+/**************************************************************************************************
+ *  Pinout
+ **************************************************************************************************/
+/* I2C */
+#define BSP_I2C_SCL      (CONFIG_BSP_I2C_GPIO_SCL)
+#define BSP_I2C_SDA      (CONFIG_BSP_I2C_GPIO_SDA)
+
+/* USB */
+#if ESP_ROM_USB_OTG_NUM > 0
+#define BSP_USB_POS      (USBPHY_DP_NUM)
+#define BSP_USB_NEG      (USBPHY_DM_NUM)
+#else
+#define BSP_USB_POS      (GPIO_NUM_NC)
+#define BSP_USB_NEG      (GPIO_NUM_NC)
+#endif
+
+/* SD card */
+#define BSP_SD_CMD       (CONFIG_BSP_SD_CMD)
+#define BSP_SD_CLK       (CONFIG_BSP_SD_CLK)
+#define BSP_SD_D0        (CONFIG_BSP_SD_D0)
+#define BSP_SD_D1        (CONFIG_BSP_SD_D1)
+#define BSP_SD_D2        (CONFIG_BSP_SD_D2)
+#define BSP_SD_D3        (CONFIG_BSP_SD_D3)
+
+/* Buttons */
+#define BSP_BUTTON_1_IO  (CONFIG_BSP_BUTTON_1_GPIO)
+#define BSP_BUTTON_2_IO  (CONFIG_BSP_BUTTON_2_GPIO)
+#define BSP_BUTTON_3_IO  (CONFIG_BSP_BUTTON_3_GPIO)
+#define BSP_BUTTON_4_IO  (CONFIG_BSP_BUTTON_4_GPIO)
+#define BSP_BUTTON_5_IO  (CONFIG_BSP_BUTTON_5_GPIO)
+
+/* Leds */
+#define BSP_LED_1_IO     (CONFIG_BSP_LED_1_GPIO)
+#define BSP_LED_2_IO     (CONFIG_BSP_LED_2_GPIO)
+#define BSP_LED_3_IO     (CONFIG_BSP_LED_3_GPIO)
+#define BSP_LED_4_IO     (CONFIG_BSP_LED_4_GPIO)
+#define BSP_LED_5_IO     (CONFIG_BSP_LED_5_GPIO)
+
+/* Buttons */
+typedef enum {
+#if CONFIG_BSP_BUTTONS_NUM > 0
+    BSP_BUTTON_1,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 1
+    BSP_BUTTON_2,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 2
+    BSP_BUTTON_3,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 3
+    BSP_BUTTON_4,
+#endif
+#if CONFIG_BSP_BUTTONS_NUM > 4
+    BSP_BUTTON_5,
+#endif
+    BSP_BUTTON_NUM
+} bsp_button_t;
+
+/* Leds */
+typedef enum {
+#if CONFIG_BSP_LEDS_NUM > 0
+    BSP_LED_1,
+#endif
+#if CONFIG_BSP_LEDS_NUM > 1
+    BSP_LED_2,
+#endif
+#if CONFIG_BSP_LEDS_NUM > 2
+    BSP_LED_3,
+#endif
+#if CONFIG_BSP_LEDS_NUM > 3
+    BSP_LED_4,
+#endif
+#if CONFIG_BSP_LEDS_NUM > 4
+    BSP_LED_5,
+#endif
+    BSP_LED_NUM
+} bsp_led_t;
+
+/* Default LED effects */
+enum {
```

Comment:
Are these symbols connected with these https://github.com/espressif/esp-iot-solution/blob/master/components/led/led_indicator/include/led_indicator_blink_default.h#L17 ?

Reply:
No, these symbols are connected with these: https://github.com/espressif/esp-bsp/blob/bsp/generic/bsp/esp_bsp_generic/src/led_blink_defaults.c I made own defaults, it is better for us.